core/trino-main/src/main/java/io/trino/connector/WorkerDynamicCatalogManager.java

```diff
@@ -45,8 +55,11 @@ public class WorkerDynamicCatalogManager
 
     private final CatalogFactory catalogFactory;
 
-    private final Lock catalogsUpdateLock = new ReentrantLock();
+    private final ReadWriteLock catalogsLock = new ReentrantReadWriteLock();
+    private final Lock catalogLoadingLock = catalogsLock.readLock();
+    private final Lock catalogRemovingLock = catalogsLock.writeLock();
     private final ConcurrentMap<CatalogHandle, CatalogConnector> catalogs = new ConcurrentHashMap<>();
+    private final ExecutorService executor = Executors.newVirtualThreadPerTaskExecutor();
```

Comment:
we are cool with the virtual thread approach yes?

Reply:
Yes!